OmsAgent/Fairfax/omsagent_ff.py

```diff
@@ -116,11 +118,29 @@ def enable(hutil):
     workspaceKey = protected_settings.get("workspaceKey")
     proxy = protected_settings.get("proxy")
     vmResourceId = protected_settings.get("vmResourceId")
+    stopOnMultipleConnections = protected_settings.get("stopOnMultipleConnections")
     if workspaceId is None:
         raise ValueError("Workspace ID cannot be None.")
     if workspaceKey is None:
         raise ValueError("Workspace key cannot be None.")
 
+    if stopOnMultipleConnections is not None and stopOnMultipleConnections is True:
+        output_file = tempfile.NamedTemporaryFile('w')
+        output_file.close()
+
+        list_exit_code = ScriptUtil.run_command(hutil, ScriptUtil.parse_args(WorkspaceCheckCommand), OmsAdminWorkingDirectory, 'Check If Already Onboarded', ExtensionShortName, hutil.get_extension_version(), False, interval = 30, output_file.name)
+
+        # If the printout includes "No Workspace" then there are no workspaces onboarded to the machine
+        # Otherwise the workspaces that have been onboarded are listed in the output
+        output_file_handle = open(output_file.name, 'r')
+        connectionExists = False
+        if "No Workspace" not in output_file_handle.read():
+            connectionExists = True
+        output_file_handle.close()
+
+        if connectionExists:
+            sys.exit("LINUXOMSAGENTEXTENSION_ERROR_MULTIPLECONNECTIONS")
```

Comment:
I wasn't sure how to return here, since we don't currently return any specific error messages in our code currently. I was referencing the python documentation [here](https://docs.python.org/2/library/sys.html#sys.exit) where you can return an error message string.